libeos-payg/manager.c

```diff
@@ -1702,12 +1702,8 @@ epg_manager_wallclock_time_changed (EpgProvider *provider,
        * CLOCK_REALTIME, but set_expiry_time expects the current timestamp in
        * terms of CLOCK_BOOTTIME, so we get the time here again. */
       guint64 now_secs = epg_clock_get_time (self->clock);
-
-      /* FIXME: This actually extends the expiration rather than reducing it
-       * but the Endless backend is basically deprecated anyway in favor of
-       * Angaza.
-       */
-      set_expiry_time (self, NULL, FALSE, now_secs, (guint64)delta);
+      set_expiry_time (self, NULL, FALSE, now_secs,
+                       self->expiry_time_secs - now_secs - (guint64)delta);
```

Comment:
If `self->expiry_time_secs - now_secs` is negative, this will underflow.

Ignoring that, if `(self->expiry_time_secs - now_secs)` is less than `delta` (i.e. the delta consumes all remaining time) then it will also underflow.

Reply:
Both cases should now be handled in the new commits.